two-phase-commit/src/Makefile

```diff
@@ -0,0 +1,35 @@
+BIN := bin/
+
+MSTR = $(BIN)tpc_master
+SLV = $(BIN)tpc_slave
+
+CFLAGS = -g -Wall
+
+MASTER_OBJ =  $(BIN)tpc_clnt.o $(BIN)tpc_xdr.o $(BIN)tpc_master.o
+SLAVE_OBJ = $(BIN)tpc_slave.o $(BIN)tpc_xdr.o $(BIN)tpc_svc.o
+
+all:
+	@rpcgen tpc.x
+	@make objects
+
+objects: $(SLV) $(MSTR)
+
+$(MSTR): $(MASTER_OBJ) tpc.h vote.h
+	@echo "compiling master"
+	gcc -o $(MSTR) $(MASTER_OBJ) -lpthread
+
+$(SLV): $(SLAVE_OBJ) tpc.h vote.h
+	@echo "compiling slave"
+	gcc -o $(SLV) $(SLAVE_OBJ)
+
+$(SLAVE_OBJ): tpc_slave.c tpc_xdr.c tpc_svc.c
+	@gcc -c -Wall -o $(BIN)tpc_slave.o tpc_slave.c
+	@gcc -c -Wall -o $(BIN)tpc_xdr.o tpc_xdr.c
```

Comment:
The rule for `bin/tcp_xdr.o` should be split out to something like `MUTUAL_OBJ` as it is listed under both `MASTER_OBJ` and `SLAVE_OBJ`, but is only compiled under the latter. This could cause compile errors if we were only compiling the necessary binaries on a node. Further, this would likely remove `make`'s warnings about "overriding commands" and "ignoring old commands" for `bin/tcp_xdr.o`.

Reply:
fixed